advanced/matrix/matrix.py

```diff
@@ -0,0 +1,220 @@
+"""An implementation of the class for the data type 'Matrix'."""
+
+
+class Matrix:
+    """A class used to represent an immutable n*n matrix.
+
+        Attributes:
+            matrix_rows (tuple): a tuple of tuples representing a matrix.
+
+        Raises:
+            TypeError: self is not a tuple of tuples.
+            ValueError: self doesn't have an equal number of rows and columns.
```

Comment:
self doesn't exists yet at this point (you should be explicit what are you talking about)